scripts/generate-project-map.py

```
@@ -0,0 +1,346 @@
+    import umap
+    from sklearn.cluster import KMeans
+    from sklearn.metrics import adjusted_rand_score
+
+    norm = mat / np.clip(np.linalg.norm(mat, axis=1, keepdims=True), 1e-9, None)
+    xy = umap.UMAP(
+        n_components=2, n_neighbors=args.neighbors, min_dist=args.min_dist,
+        metric='cosine', random_state=args.seed,
+    ).fit_transform(norm)
+    xy = np.asarray(xy, dtype=np.float64)
+
+    print(f'[4/4] クラスタリング（KMeans k={args.clusters}）')
```

Ruff (0.15.21)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)